### ISSUE TITLE
ECRPushImage warning in Azure Devops due to Node10 dependency

### ISSUE THREAD

**Issue by ganeshranfignum** [aws/aws-toolkit-azure-devops#566]
The ECRPushImage Task uses Node10 which is no longer supported . This causes a warning to be displayed on our build process in Azure Devops

`##[warning]Task 'Amazon ECR Push' version 1 (ECRPushImage@1) is dependent on a Node version (10) that is end-of-life. Contact the extension owner for an updated version of the task. Task maintainers should review Node upgrade guidance: https://aka.ms/node-runner-guidance
`

**Comment by jbstans** [comment]
This isn't just limited to ECR - we're getting the same on AWSShellScript:

```##[warning]Task 'AWS Shell Script' version 1 (AWSShellScript@1) is dependent on a Node version (10) that is end-of-life. Contact the extension owner for an updated version of the task. Task maintainers should review Node upgrade guidance: https://aka.ms/node-runner-guidance```

I suspect the whole toolkit needs a v2 with a node update.

**Comment by tmrmotta** [comment]
It is also happening on the s3 upload

`##[warning]Task 'Amazon S3 Upload' version 1 (S3Upload@1) is dependent on a Node version (10) that is end-of-life. Contact the extension owner for an updated version of the task. Task maintainers should review Node upgrade guidance: https://aka.ms/node-runner-guidance
`

**Comment by mhvelplund** [comment]
AWS support ADO seems pretty dead. I wish I was stuck with ADO either :/

**Comment by mhvelplund** [comment]
I found out why they don't just bump the Node version. From PR #560:

> Supporting node 16 requires bumping azure-pipelines-task-lib. We are currently blocked from doing this because this version bump uses a newer version of shelljs as a dependency, which produces runtime errors for our task. (see PR description in https://github.com/aws/aws-toolkit-azure-devops/pull/539 for more details).

**Comment by decampo04** [comment]
Just to add, also seeing this on SystemsManagerGetParameter@1 task too:

`##[warning]Task 'AWS SSM Get Parameter' version 1 (SystemsManagerGetParameter@1) is dependent on a Node version (10) that is end-of-life. Contact the extension owner for an updated version of the task. Task maintainers should review Node upgrade guidance: https://aka.ms/node-runner-guidance
`

**Comment by seanezell** [comment]
Just to add to the pile, we are seeing this on S3Upload, AWSShellScript. 

We are also seeing the same on Terraform and Terraform-outputs. Different extensions, but the same issue.

**Comment by JeroenBodeTriple** [comment]
Same issue here for the AWSCLI, AWSShellScript, LambdaDeployFunction and ECRPushImage on our end.
My guess is all the tasks for the aws-toolkit-azure-devops are affected by this.

**Comment by symnatbay** [comment]
Seeing the same over in our org. No new information, just leaving this comment to help express the potential impact of this deprecation.

**Comment by sameerahmed720** [comment]
Its a on going issue with AWS and Terraform , fixed terraform issue by upgrading its version 
But AWSShellScript , AWSCLI issue is consistent even after version upgrade

**Comment by isarastov** [comment]
I'm seeing the same for S3Upload@1, SendMessage@1

**Comment by tayalnishu** [comment]
Getting the same error for all tasks used from `aws-toolkit-azure-devops`.  Would be good to know the timeline when it is expected to be upgraded.

**Comment by rynop** [comment]
Anyone have an official Microsoft doc that states when they are pulling the plug on node v10?  

I opened an AWS support ticket and I can tell you AWS is aware of this.

**Comment by saurabhajmera** [comment]
I like to acknowledge that we are aware of this problem. In our past attempts to address this, we got blocked by [a shelljs related JS bundler issue](https://github.com/microsoft/azure-pipelines-task-lib/issues/942) which prevented us from updating `azure-pipelines-task-lib`. We are looking into this again and will post an update once we have more information.

**Comment by adurangalvis** [comment]
> I like to acknowledge that we are aware of this problem. In our past attempts to address this, we got blocked by [a shelljs related JS bundler issue](https://github.com/microsoft/azure-pipelines-task-lib/issues/942) which prevented us from updating `azure-pipelines-task-lib`. We are looking into this again and will post an update once we have more information.

Is there a timeline when this is planned to be resolved?

**Comment by saurabhajmera** [comment]
> Is there a timeline when this is planned to be resolved?

Not yet, but we will share more information once we have an update. Thank you for your patience.

**Comment by ajith-shan-m** [comment]
Hey @saurabhajmera

Is there any updates regarding this ongoing issue ?

**Comment by saurabhajmera** [comment]
@ajith-shan-m, We are actively working on a short-term solution, which we plan to release in the coming weeks. Addressing the shelljs issue in the long term, however, will require changes to the `azure-pipelines-task-lib` library, which is maintained by Microsoft. While we are in ongoing communication with Microsoft about this, we currently do not have a timeline for when the long-term fix will be available.

**Comment by hayemaxi** [comment]
We have updated the tasks to support Node 20 as per [microsoft guidance](https://learn.microsoft.com/en-us/azure/devops/extend/develop/add-build-task?view=azure-devops#q-how-can-i-upgrade-my-custom-task-to-the-latest-node). It is now released under version 1.18.0

**Comment by rynop** [comment]
I can confirm the warning has gone away. Thanks!

**Comment by mateofd** [comment]
Same here! Thanks!

**Comment by Hektor-V** [comment]
Also confirmed, thank you!

**Comment by isarastov** [comment]
The warning is gone in 1.18.0, thank you!

**Comment by ajith-shan-m** [comment]
The warning is gone now! Thank you

**Comment by mingo-eric** [comment]
I can see in my pipeline that it lists 1.18

```
Task         : Amazon S3 Upload
Description  : Upload file and folder content to an Amazon Simple Storage Service (S3) Bucket on AWS
Version      : 1.18.0
```

But I still seeing it using `node10\bin\node.exe` in an error we are facing. Is this change dependent on anything other than just having the newest version of the task?

**Comment by hayemaxi** [comment]
> I can see in my pipeline that it lists 1.18
> 
> ```
> Task         : Amazon S3 Upload
> Description  : Upload file and folder content to an Amazon Simple Storage Service (S3) Bucket on AWS
> Version      : 1.18.0
> ```
> 
> But I still seeing it using `node10\bin\node.exe` in an error we are facing. Is this change dependent on anything other than just having the newest version of the task?

The change allows agents to use node 20 to run the tasks, so it seems likely that your agent either does not have node 20 installed or it is not the default. AFAIK, in most cases microsoft-hosted agents should already have node20 installed and used by default, while on-prem agents will need to be manually updated.

Can you verify the agent it is running on has node 20 and it's the default? Try using the Node JS installer task in your pipeline before running the AWS tasks. If you still see this, please open a new issue.
